### ISSUE TITLE
No Access to Gradients of Exported Model

### ISSUE THREAD

**Issue by guentj** [keras-team/autokeras#1672]
### Bug Description

I am trying to use automatic differentiation from TensorFlow. In order to do this, I created a model with AutoKeras and exported it as a TensorFlow model.

When trying to get any gradients afterwards, they are reported to be 'None'. Models that were created manually in tensorflow are able to get gradients with the code below.

### Bug Reproduction

Code for reproducing the bug:

```
import tensorflow as tf
import autokeras as ak

(x_train, y_train), (x_test, y_test) = tf.keras.datasets.mnist.load_data()
clf = ak.ImageClassifier(overwrite = True,max_trials = 1)
clf.fit(x_train, y_train, epochs = 1)
model = clf.export_model()

x_tensor = tf.convert_to_tensor(x_test)
with tf.GradientTape() as g:
    g.watch(x_tensor)
    y_pred = model(x_tensor)
    dx = g.gradient(y_pred,x_tensor)
    print(dx)

```

### Expected Behavior

The gradients should be accessible

### Setup Details

Include the details about the versions of:

- OS type and version: Windows 11 10.0.22000
- Python: 3.9.0
- autokeras: 1.0.16
- keras-tuner: 1.0.4
- numpy: 1.19.3
- tensorflow: 2.5.0

**Comment by guentj** [comment]
The problem is that the format required for training is float32 and the default data format for training is uint8, Newer versions of tensorflow also give warnings about this problem. The following code works: 

```
import tensorflow as tf
import autokeras as ak
import numpy as np 

(x_train, y_train), (x_test, y_test) = tf.keras.datasets.mnist.load_data()
clf = ak.ImageClassifier(overwrite = True,max_trials = 1)

x_train_float = x_train.astype(np.float32)
clf.fit(x_train_float, y_train, epochs = 1)
model = clf.export_model()

x_tensor = tf.convert_to_tensor(x_test.astype(np.float32))
with tf.GradientTape() as g:
    g.watch(x_tensor)
    y_pred = model(x_tensor)
    dx = g.gradient(y_pred,x_tensor)
    print(dx)

```
